Sources/TensorFlow/Core/Serialization.swift

```diff
@@ -67,16 +64,30 @@ public class TensorFlowCheckpointReader {
         return TensorShape([Int64](dimensionsBufferPointer).map(Int.init))
     }
 
-    /// Returns the data type of the tensor with the provided name stored in the checkpoint.
-    @inlinable
-    public func dataType(ofTensorNamed name: String) -> TensorDataType {
-        TensorDataType(TF_CheckpointReaderGetVariableDataType(handle, name))
+    /// Returns the type of the tensor with the provided name stored in the checkpoint.
+    public func typeOfTensor(named name: String) -> Any.Type {
```

Comment:
```suggestion
    public func scalarTypeOfTensor(named name: String) -> Any.Type {
```

Reply:
Done.

Reply:
Oh actually you can make it more specific: return `TensorFlowScalar.self`.

Reply:
That does not work because of the `String` type.